src/Thruway/Role/Subscriber.php

```diff
@@ -219,13 +251,49 @@ public function subscribe(ClientSession $session, $topicName, callable $callback
             "options"    => $options,
             "deferred"   => $deferred
         ];
-
+        
         array_push($this->subscriptions, $subscription);
-
+        
         $subscribeMsg = new SubscribeMessage($requestId, $options, $topicName);
         $session->sendMessage($subscribeMsg);
 
         return $deferred->promise();
     }
-
+	
+    /**
+     * process unsubscribe
+     * @param ClientSession $session
+     * @param string $subscriptionId
+     * @return boolean
+     */
+    public function unsubscribe(ClientSession $session, $subscriptionId)
+    {
+        $requestId = Utils::getUniqueId();
+        $subscriptionExists = false;
+        $deferred  = new Deferred();
+        
+        foreach ($this->subscriptions as $i => $subscription) {
+            if ($subscription["subscription_id"] == $subscriptionId) {
+                $subscriptionExists = true;
+                
```

Comment:
What do you think is the best way to achieve this? We could set the `subscription["callback"]` to `null`, requiring us to update the `processEvent` method to handle this situation. Or maybe delete the subscription from the subscriptions list, but we have to deal with the promise that must be resolved or rejected when the router responds.